verifyFiles.py

```diff
@@ -132,17 +133,44 @@ def check_files(conditions_dict):
 # For instance:
 #  {'condition1 vs condition2': {'c/c.txt': 0, 'a.txt': 2}}
 #  means that 'c/c.txt' is identical for all subjects in conditions condition1 and condition2 while 'a.txt' differs in two subjects.
-def n_differences_across_subjects(conditions_dict,root_dir,metrics,checksums_from_file_dict,checksum_after_file_path,check_corruption):
+def n_differences_across_subjects(conditions_dict,root_dir,metrics,checksums_from_file_dict,checksum_after_file_path,check_corruption,sqlite_db_path):
     # For each pair of conditions C1 and C1
     product = ((i,j) for i in conditions_dict.keys() for j in conditions_dict.keys())
     diff={} # Will be the return value
     metric_values={}
     path_names = conditions_dict.values()[0].values()[0].keys()
+    #dictionary_checksum is used for storing the computed checksum values and to avoid computing the checksums for the files multiple times
+    dictionary_checksum={}
+    #dictionary_executables is used for tracking the files that we have already found the executables for
+    dictionary_executables={}
+    #Initialize sqlite connection(Not sure if it is the best way)
+    if sqlite_db_path:
+      conn = sqlite3.connect(sqlite_db_path)
     # Go through all pairs of conditions
     for c, d in product:
         if c < d: # Makes sure that pairs are not ordered, i.e. {a,b} and {b,a} are the same
             key=c+" vs "+d
             diff[key]={}
+            # if c and d both start with x-RUN-y (same x, different
+            # y), then assume that they are different runs from the
+            # same condition. In this case, if there are differences
+            # for a given file name below, the reprozip trace should
+            # be inspected to determine the executable that created
+            # such inter-run differences in the same condition. Also,
+            # print a log_info saying "Identified c1 and c2 as two
+            # different runs of the same condition".
+	    is_intra_condition_run=False
+	    if "RUN" in c and "RUN" in d:
```

Comment:
You should be as specific as possible, e.g., check that the condition name matches something like `.*-RUN-[0-9]*`.